isisd/isis_spf.c

```diff
@@ -1296,9 +1304,21 @@ static int isis_spf_preload_tent_ip_reach_cb(const struct prefix *prefix,
 			break;
 		}
 	}
-	if (!has_valid_psid)
-		isis_spf_add_local(spftree, vtype, &ip_info, NULL, 0, NULL,
-				   parent);
```

Comment:
Same comment here. This avoid to duplicate code.

Reply:
Same comment here. But transition metric requires an additional call to isis_spf_add_local() to create the vertex.
```
	if (!has_valid_psid) {
		isis_spf_add_local(spftree, vtype, &ip_info, NULL, 0, NULL,
				   parent);
		if (transition)
            		isis_spf_add_local(spftree, 
                               		   VTYPE_IPREACH_TE, 
                               		   &ip_info, NULL, 0,
					   NULL, parent);
	}
```